src/Lexer.php

```diff
@@ -360,17 +361,17 @@ private function lexExpression(): void
         // punctuation
         elseif (str_contains(self::PUNCTUATION, $this->code[$this->cursor])) {
             // opening bracket
-            if (str_contains('([{', $this->code[$this->cursor])) {
+            if (str_contains('{', $this->code[$this->cursor])) {
                 $this->brackets[] = [$this->code[$this->cursor], $this->lineno];
             }
             // closing bracket
-            elseif (str_contains(')]}', $this->code[$this->cursor])) {
+            elseif (str_contains('}', $this->code[$this->cursor])) {
```

Comment:
Do we still ensure we have proper balanced brackets for each type of brackets in the new logic ?

Reply:
Fixed, with some new tests to cover the code.